templates/main/index.html.ep

```diff
@@ -22,24 +22,13 @@
   </div>
 </div>
 
-<div id="build-results"></div>
-% for my $groupresults (@$results) {
-    % my $group              = $groupresults->{group};
-    % my $build_results      = $groupresults->{build_results};
-    % my $max_jobs           = $groupresults->{max_jobs};
-
-    % if (@{$groupresults->{children}}) {
-        <h2>
-            %= link_to $group->{name} => url_for('parent_group_overview', groupid => $group->{id})
-        </h2>
-        %= include 'main/group_builds', build_results => $build_results, group => $group, max_jobs => $max_jobs, children => $groupresults->{children}, default_expanded => $default_expanded
-    % } else {
-        <h2>
-            %= link_to $group->{name} => url_for('group_overview', groupid => $group->{id})
-        </h2>
-        %= include 'main/group_builds', build_results => $build_results, group => $group, max_jobs => $max_jobs, children => undef, default_expanded => 0
-    % }
-% }
+<div id="build-results-loading">
+    <div class="d-flex justify-content-center">
+        <i class="fas fa-circle-notch fa-spin fa-4x" role="status"><span class="sr-only">Loading…</span></i>
+    </div>
+</div>
+<div id="build-results" data-build-results-url="<%= url_for('dashboard_build_results') %>">
+</div>
 
 <div class="card card-outline-secondary filter-panel-bottom" id="filter-panel">
```

Comment:
can you move that filter box also into the sub page? Not for performance reasons but because then the sub-page would also be usable on it's own

Reply:
> then the sub-page would also be usable on it's own

It wouldn't be. The "sub page" is only an HTML fragment and not a complete page. The form is actually used by the JavaScript code to conveniently make the AJAX query. If it wasn't present here and instead be part of the AJAX response I needed to provide a different implementation for the initial AJAX query and the form. That makes things needlessly complicated.